pinot-core/src/main/java/org/apache/pinot/core/transport/QueryRouter.java

```diff
@@ -114,6 +115,7 @@ public AsyncQueryResponse submitQuery(long requestId, String rawTableName,
     Map<ServerRoutingInstance, InstanceRequest> requestMap = new HashMap<>();
     if (offlineBrokerRequest != null) {
       assert offlineRoutingTable != null;
+      BrokerRequestToQueryContextConverter.validateGapfillQuery(offlineBrokerRequest);
```

Comment:
Let's not validate gapfill here as it will add overhead to all queries (converting broker request to query context)

Reply:
Fixed